File: adrop-ads-example-ios-objective-c/QuestBannerViewController.m
//
//  QuestBannerViewController.m
//  adrop-ads-example-ios-objective-c
//
//  Created by Leo Kim on 4/14/25.
//

#import "QuestBannerViewController.h"

@interface QuestBannerViewController ()

@property (weak, nonatomic) IBOutlet UIView *feedContainerView;
@property (weak, nonatomic) IBOutlet UIView *coverContainerView;

@property (strong, nonatomic) AdropQuestBanner *feedBanner;
@property (strong, nonatomic) AdropQuestBanner *coverBanner;

@end

@implementation QuestBannerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"Quest Banner";
    
    self.feedBanner = [[AdropQuestBanner alloc] initWithChannel:@"adrop" format:AdropQuestBannerFormatFEED];
    self.feedBanner.frame = CGRectMake(0, 0, self.view.bounds.size.width, 265);
    self.feedBanner.delegate = self;
    [self.feedBanner load];
    [self.feedContainerView addSubview:self.feedBanner];
    
    self.coverBanner = [[AdropQuestBanner alloc] initWithChannel:@"adrop" format:AdropQuestBannerFormatCOVER];
    self.coverBanner.frame = CGRectMake(0, 0, self.view.bounds.size.width, 74);
    self.coverBanner.delegate = self;
    [self.coverBanner load];
    [self.coverContainerView addSubview:self.coverBanner];
}

#pragma mark - AdropBannerDelegate

- (void)onAdReceived:(AdropBanner *)banner {
    if (banner == self.feedBanner) {
        NSLog(@"onAdReceived FEED_BANNER");
    } else if (banner == self.coverBanner) {
        NSLog(@"onAdReceived COVER_BANNER");
    }
}

- (void)onAdFailedToReceive:(AdropBanner *)banner :(AdropErrorCode)errorCode {
    if (banner == self.feedBanner) {
        NSLog(@"onAdFailedToReceive FEED_BANNER");
    } else if (banner == self.coverBanner) {
        NSLog(@"onAdFailedToReceive COVER_BANNER");
    }
}

@end

File: adrop-ads-example-ios/QuestBannerViewController.swift
//
//  QuestBannerViewController.swift
//  adrop-ads-example-ios
//
//  Created by Leo Kim on 4/14/25.
//

import UIKit
import AdropAds

class QuestBannerViewController: UIViewController {
    @IBOutlet weak var feedContainerView: UIView!
    @IBOutlet weak var coverContainerView: UIView!
    
    var feedBanner: AdropQuestBanner!
    var coverBanner: AdropQuestBanner!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        title = "Quest Banner"
        
        feedBanner = AdropQuestBanner(channel: "adrop", format: .FEED)
        feedBanner.frame = CGRect(x: 0, y: 0, width: view.bounds.width, height: 265)
        feedBanner.delegate = self
        feedBanner.load()
        feedContainerView.addSubview(feedBanner)
        
        coverBanner = AdropQuestBanner(channel: "adrop", format: .COVER)
        coverBanner.frame = CGRect(x: 0, y: 0, width: view.bounds.width, height: 74)
        coverBanner.delegate = self
        coverBanner.load()
        coverContainerView.addSubview(coverBanner)
    }
}

extension QuestBannerViewController: AdropBannerDelegate {
    func onAdReceived(_ banner: AdropAds.AdropBanner) {
        if (banner == feedBanner) {
            print("onAdReceived FEED_BANNER")
        } else if (banner == coverBanner) {
            print("onAdReceived COVER_BANNER")
        }
    }
    
    func onAdFailedToReceive(_ banner: AdropAds.AdropBanner, _ errorCode: AdropAds.AdropErrorCode) {
        if (banner == feedBanner) {
            print("onAdFailedToReceive FEED_BANNER")
        } else if (banner == coverBanner) {
            print("onAdFailedToReceive COVER_BANNER")
        }
    }
}
